src/slm/ids2ngram/ids2ngram.cpp

```diff
@@ -181,18 +181,22 @@ main(int argc, char* argv[])
     while (optind < argc) {
         printf("Processing %s:", argv[optind]); fflush(stdout);
         FILE *fp = fopen(argv[optind], "rb");
-        switch (N) {
-        case 1:
-            ProcessingRead<1>(fp, swap, para_offsets, paraMax);
-            break;
-        case 2:
-            ProcessingRead<2>(fp, swap, para_offsets, paraMax);
-            break;
-        case 3:
-            ProcessingRead<3>(fp, swap, para_offsets, paraMax);
-            break;
+        if (fp != NULL) {
```

Comment:
could also put
```c++
if (!fp) {
  int err = errno;
  cerr << "failed to open " << argv[optind] << ": " << strerror(err) << std::endl;
  ++optind;
  continue;
}
```


1. to reduce the indent level
1. to have less code churn
1. to print out the reason why `fopen()` failed.

Reply:
I think we'd better keep the condition block, otherwise you'll have to write `++optind` twice which is not good for code maintance.